packages/adblocker/src/data-view.ts

```diff
@@ -302,76 +314,76 @@ export default class StaticDataView {
   }
 
   public pushNetworkRedirect(str: string): void {
-    if (this.enableCompression === true) {
-      this.pushBytes(compression.deflateNetworkRedirectString(str));
+    if (this.compression !== null) {
```

Comment:
isn't `false` a better falsy value?

Reply:
I think the name of the attribute might be a bit misleading, but its type is either `null` or `Compression` (an instance of the class), so I use `null` as a way to indicate that the instance of the class does not exist and would prefer to keep types stable. Probably the attribute could be renamed in something like `compressor`?

Reply:
if so why not to check for truthiness `if (this.compression) {` ?

Reply:
It's half a personal style convention and half-based on a performance argument. In general I try to be as explicit as possible about my intents to both express more information from the code and give more information to the JavaScript engine. In this case I know it's either `null` or an instance of `Compression` so I check explicitly instead of giving more work to the engine to figure out what "truthy" would mean in this context (not saying it has a noticeable impact on performance here; but I consider it a good practice at this point). I might be wrong, but it seems to me that the "check for any truthy value" in JavaScript is a convenience because we are never 100% sure what we get without a good type system; in TypeScript we have much more confidence there.